### ISSUE TITLE
Persistent TOC

### ISSUE THREAD

**Issue by lloyd094** [hang-up/buuk#22]
We've previously tried to get TOC (table of contents) on Buk, but found out it causes an issue. Since then, I've had an idea that I think would improve Buk.

The sidebar of Buk is built off the documents in the Wiki - what if there was another level to that sidebar (for the document itself) where a TOC is generated off of set headers (say, the option for H1 and H2 only, or H2, H3, and H4 only) that could be configured via the bootstrap or manifest. This way, when using a large document, the user simply goes to the left menu where everything else is located, and clicks on the 'button' that links to that specific header, and the page scrolls to that header.

Personally, I think this would be much more useful than a traditional TOC, but whether it's feasible or not is another question.

**Comment by hang-up** [comment]
That's an amazing suggestion!
 It would require to pretty much rework the entire parsing logic though... I can't put a timeline on this feature as of right now but if you want to collaborate and get a head start on this feature, I'll gladly put you as a collaborator on Buk!

**Comment by lloyd094** [comment]
I've spent the past couple of hours looking into the subject - in terms of collaboration, I'm probably not the person you're looking for. While I plan to be using Buk for the foreseeable future, I will be starting Law School in the next month. Additionally, I know very little about JavaScript, but not enough to make anything. That being said, I would like to help as much as possible - and the lack of a timeline is fine considering what's being asked.

I've researched a bit into Materialize and your structuring for the sidebar, and I think there's three main things that are required for this feature (based on what I know):

1. During `npm run build`, there needs to be something that runs in the background ([markdown-toc](https://github.com/jonschlinkert/markdown-toc) possibly?) that can export a .json (or series of .json files) based off of the .md documents in `/docs`.

2. Update the template in `/src/js/components/sidebar/sidebar-item.vue` to support generating a series of "sub-categories" from the TOC's listed in the .json file(s), each based on the .md file being used at the time. This of course is used in the `npm run build` process as well, after the generating of the .json files.

3. Finally, we would need a way to add internal links from the sidebar to the page currently displayed, so it can navigate directly to the header based off the TOC. Not sure if it will help for how you have Buk setup, but this link might be helpful here: [SitePoint](https://www.sitepoint.com/scroll-smoothly-javascript/)

I wish I could be more help on the matter in terms of being able to code some of this, but I hope something from this might have helped. If there are things of a more narrow focus that you'd like help on, let me know and I'll see what I can provide.

Also, thanks again for Buk - it's really phenomenal, and it's fun to see it grow!
